### ISSUE TITLE
GROUP BY <table> does not use table alias

### ISSUE THREAD

**Issue by SebastianS90** [jOOQ/jOOQ#13751]
### Expected behavior

Since jOOQ 3.17, `Table<R> extends SelectField<R>, GroupField` (great addition, by the way!).
When using `.groupBy(x)` where `x` is an aliased table, the generated query should use the alias name in the `group by` clause.

### Actual behavior
The generated query uses the original table name.

### Steps to reproduce the problem

[MCVE](https://github.com/SebastianS90/jOOQ-mcve/tree/groupby-alias/jOOQ-mcve-java)

**Schema:**
```sql
CREATE SCHEMA mcve;

CREATE TABLE mcve.author (
  id INT NOT NULL,
  name VARCHAR,

  CONSTRAINT pk_author PRIMARY KEY (id)
);

CREATE TABLE mcve.book (
  id INT NOT NULL,
  author_id INT NOT NULL,
  name VARCHAR,

  CONSTRAINT pk_book PRIMARY KEY (id),
  CONSTRAINT fk_book_author FOREIGN KEY (author_id) REFERENCES author (id)
);

INSERT INTO mcve.author (id, name)
VALUES (1, 'John Doe'), (2, 'Jane Doe');

INSERT INTO mcve.book (id, author_id, name)
VALUES (1, 1, 'Foo'), (2, 1, 'Bar'), (3, 2, 'Foobar');
```

**Test:**
```java
Author a = AUTHOR.as("a");
Book b = BOOK.as("b");
var authorBooks = ctx
        .select(a, DSL.arrayAgg(b.NAME))
        .from(a)
        .innerJoin(b).onKey(Keys.FK_BOOK_AUTHOR)
        .groupBy(a)
        .fetchMap(r -> r.value1().getId(), r -> Set.of(r.value2()));

assertEquals(Set.of("Foo", "Bar"), authorBooks.get(1));
assertEquals(Set.of("Foobar"), authorBooks.get(2));
```

**Generated query:**
```sql
 select "a"."ID" "a.ID", "a"."NAME" "a.NAME", array_agg("b"."NAME") from "MCVE"."AUTHOR" "a" join "MCVE"."BOOK" "b" on "b"."AUTHOR_ID" = "a"."ID" group by "MCVE"."AUTHOR"."ID"
```
**Error:** `Column "MCVE.AUTHOR.ID" not found`

Without the alias `.as("a")` it works as expected.

### Versions

- jOOQ: 3.17.1
- Java: 17
- Database (include vendor): H2
- OS: Linux

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce this. Will look into it right away.

**Comment by lukaseder** [comment]
Ah yes, this could be seen as another side effect of the underlying issue of https://github.com/jOOQ/jOOQ/issues/12214, where aliased tables don't have alias sensitive key information.

The problematic logic is in `GroupFieldList`. The problem only appears when the `pk.getFieldsArray()` method is invoked, i.e.

- There is a primary key
- Projecting functional dependencies of key columns is supported

```java
    private final Field<?>[] fields(Context<?> ctx, Table<?> t) {
        UniqueKey<?> pk = t.getPrimaryKey();

        if (pk == null || NO_SUPPORT_GROUP_FUNCTIONAL_DEP.contains(ctx.dialect()))
            return t.fields();
        else
            return pk.getFieldsArray();
    }
```

For example, the issue does not manifest in Oracle.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.2 (https://github.com/jOOQ/jOOQ/issues/13754), and 3.16.8 (https://github.com/jOOQ/jOOQ/issues/13755). Thanks again for your report!
